### ISSUE TITLE
Fine-tuning have an issue with unknown characters ?

### ISSUE THREAD

**Issue by PonteIneptique** [mittagessen/kraken#393]
I have ran a finetuning with two similar datasets, one much bigger than the other, but with a few new (frequent) characters.

Rebuilding the alphabet does seem to break anything:

```
[10/08/22 18:09:59] WARNING  alphabet mismatch: chars in training set only: {'=', 'ᵖ', '́', 'Z', '⁋', 'đ', '¬', '+', '℥', '̈', '÷', 'Y', '\uf038', '\uf2da', "'", '*', 'J'} (not included in accuracy    train.py:307
                             test during training)                                                                                                                                                                 
                    WARNING  alphabet mismatch: chars in validation set only: {'ι'} (not trained)                                                                                                      train.py:311
```

BUT I get those during the first few batch of training, which would seem to be typical issues of non learnt characters:

```
                   WARNING  Non-encodable sequence  ut... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence ᵈ voc... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence ᷝ ͧ̃ ... encountered. Advancing one code point.                                                                                                       codec.py:131
                    WARNING  Non-encodable sequence  ide... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence ꝭ leg... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence ꝭ nõ... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence ... encountered. Advancing one code point.                                                                                                        codec.py:131
                    WARNING  Non-encodable sequence ̵. Tr... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence  ꝙͣu... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence  ño... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence  phi... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence ᵈ ͣ n... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence ꝵ arc... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence ꝵ ẽ ... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence ꝵ. an... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence ℥.ii.... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence  cet... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence  egͥ... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence . si... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence  ut ... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence  eiᷤ... encountered. Advancing one code point.                                                                                                     codec.py:131
                    WARNING  Non-encodable sequence ᷝ ͤ y... encountered. Advancing one code point.                                                                                                      codec.py:131
                    WARNING  Non-encodable sequence ꝭ et ... encountered. Advancing one code point.                                                                                                    codec.py:131
                    WARNING  Non-encodable sequence ... encountered. Advancing one code point.                                                                                                        codec.py:131
```

**Comment by PonteIneptique** [comment]
Damn, I just forgot to use `--resize`...
Time to go on vacations I suppose..

**Comment by mittagessen** [comment]
On 22/10/08 04:24PM, Thibault Clérice wrote:
> Closed #393 as completed.

Still a bug. It should fail if you don't specify `--resize`. Probably
because I screwed up the dataset refactor and worked on an older version
of it. I'll look into it.
